### PR TITLE
feat: support heartbeat callback and add node types

### PR DESCRIPTION
**Pull request by kallydev**
Resolved https://github.com/RSS3-Network/Global-Indexer/issues/141.

### CODE REVIEW

File: schema/node.go [hunk 1]
```diff
@@ -33,6 +33,7 @@ type Node struct {
 	APY                    decimal.Decimal        `json:"apy"`
 	ActiveScore            decimal.Decimal        `json:"activeScore"`
 	ReliabilityScore       decimal.Decimal        `json:"reliabilityScore"`
+	Type                   string                 `json:"type"`
```
Comment (brucexc) [inline schema/node.go:36]: 'Type' field has made the 'Alpha' field redundant.

File: internal/database/dialer/cockroachdb/migration/20240425092038_add_node_type_field_for_node_info_table.sql [hunk 2]
```diff
@@ -0,0 +1,10 @@
+-- +goose Up
+-- +goose StatementBegin
+ALTER TABLE "node_info"
+    ADD COLUMN "type" TEXT;
+-- +goose StatementEnd
+
+-- +goose Down
+-- +goose StatementBegin
+ALTER TABLE "node_info" DROP COLUMN "type";
```
Comment (brucexc) [inline internal/database/dialer/cockroachdb/migration/20240425092038_add_node_type_field_for_node_info_table.sql:9]: Should an index be added to the 'type' column?

File: internal/service/hub/handler/nta/node_register.go [hunk 3]
```diff
@@ -363,6 +375,41 @@ func (n *NTA) checkSignature(_ context.Context, address common.Address, message
 	return nil
 }
 
+// checkEndpoint checks if the endpoint is available and contains the node's address.
+func (n *NTA) checkEndpoint(ctx context.Context, endpoint string, address common.Address) error {
+	httpRequest, err := http.NewRequestWithContext(ctx, http.MethodGet, fmt.Sprintf("http://%s:81", endpoint), nil)
```
Comment (brucexc) [inline internal/service/hub/handler/nta/node_register.go:380]: 1. should import the HTTP provider to replace it with the fetch method.
2. the node endpoint has already had a schema added during registration. 
https://github.com/RSS3-Network/Global-Indexer/blob/main/internal/service/hub/handler/nta/node_register.go#L120

File: schema/node.go (hunk 1 above)
Comment (polebug) [reply schema/node.go:36]: @brucexc 'Alpha' field is set by the smart contract and depends on timestamp. These two fields have different meanings.

**Review by brucexc — DISMISSED** [review]

**Review by polebug — DISMISSED** [review]

**Review by polebug — APPROVED** [review]

**Pull request merged**